frontend/src/app/chart/chart.component.ts

```diff
@@ -0,0 +1,255 @@
+import { Component, OnInit, ViewEncapsulation } from '@angular/core';
+import { GetDataService } from '../services/get-data.service';
+import * as d3 from 'd3';
+
+@Component({
+  selector: 'main-chart',
+  templateUrl: './chart.component.html',
+  styleUrls: ['./chart.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class ChartComponent implements OnInit {
+  records: any;
+  loading = false;
+
+  // Chart size constants
+  chart_width = 900;
+  chart_height = 700;
+  chart_padding = 50;
+  time_parse = d3.timeParse('%s');
+  time_format = d3.timeFormat('%m/%d-%H:%M');
+
+  constructor(private service: GetDataService) {}
+
+  ngOnInit(): void {
+    this.loading = true;
+    this.service.getRecords().subscribe((response) => {
+      this.records = response.body;
+      this.loading = false;
+      this.createLineChart();
+    });
+  }
+
+  createLineChart() {
+    const records = this.records;
+    const time_parse = this.time_parse;
+    const time_format = this.time_format;
+
+    // Chart SVG Component
+    const svg = d3
+      .select('#chart-component')
+      .append('svg')
+      .attr('width', this.chart_width)
+      .attr('height', this.chart_height);
+
+    // Add a clipPath: everything out of this area won't be drawn.
+    svg
+      .append('defs')
+      .append('svg:clipPath')
+      .attr('id', 'clip')
+      .append('svg:rect')
+      .attr('width', this.chart_width - this.chart_padding * 2)
+      .attr('height', this.chart_height)
+      .attr('x', this.chart_padding)
+      .attr('y', 0);
+
+    // Create the chart variable: where both the chart and the brush take place
+    const chart = svg.append('g').attr('clip-path', 'url(#clip)');
+
+    // Scales
+    const x_scale = d3
+      .scaleTime()
+      .domain([
+        d3.min(records, (d) => this.time_parse(d[0])),
+        d3.max(records, (d) => this.time_parse(d[0])),
+      ])
+      .range([this.chart_padding, this.chart_width - this.chart_padding]);
+
+    const y_scale = d3
+      .scaleLinear()
+      .domain([0, d3.max(records, (d) => +d[1])])
+      .range([this.chart_height - this.chart_padding, this.chart_padding]);
+
+    // Create Axis
+    const x_axis = svg
+      .append('g')
+      .classed('x-axis', true)
+      .attr(
+        'transform',
+        `translate(0, ${this.chart_height - this.chart_padding})`
+      )
+      .call(
+        d3.axisBottom(x_scale)
+        // .tickFormat(this.time_format)
+      );
+    const y_axis = svg
+      .append('g')
+      .classed('y-axis', true)
+      .attr('transform', `translate(${this.chart_padding},0)`)
+      .call(d3.axisLeft(y_scale));
+
+    // Draw Area under the curve
+    const area = d3
+      .area()
+      .defined((d) => +d[1] >= 0)
+      .x((d: any) => x_scale(this.time_parse(d[0])))
+      .y0(() => y_scale.range()[0])
+      .y1((d: any) => y_scale(+d[1]));
```

Comment:
I am a bit confused about what does y0 and y1 stand for. 2 y axis? Can you explain a bit at here? Thank you.

Reply:
Here y0 y1 are used to define the area. So that only the region under the lines would be colored. 
y_scale.range()[0] means the bottom of the chart (x-axis)